llvm/tools/llvm-reduce/deltas/ReduceAliases.h

```diff
@@ -17,8 +17,8 @@
 #include "Delta.h"
 
 namespace llvm {
-void reduceAliasesDeltaPass(TestRunner &Test);
-void reduceIFuncsDeltaPass(TestRunner &Test);
+void reduceAliasesDeltaPass(TestRunner &Test, StringRef PassMessage);
```

Comment:
Something is backwards here. The delta pass implementations shouldn't need to be told what the message is. Instead, can we have these reduce*DeltaPasses return the function_ref to the implementation function, and have the driver in DeltaManager handle the printing of the message from the DeltaPass information?

Reply:
That is, the DeltaPass struct would contain function_ref<void(Oracle &, ReducerWorkitem &)> as the implementation function

Reply:
Returning the function_ref is an unnecessary indirection. The entry point signatures should just change to void(Oracle &, ReducerWorkitem&). The table can directly refer to those